lib/ansible/modules/network/onyx/onyx_wjh.py

```diff
@@ -0,0 +1,205 @@
+                        group=dict(choices = ['all', 'forwarding', 'acl']),
+                        enabled = dict(type = 'bool'),
+                        auto_export = dict(type = 'bool'),
+                        export_group = dict(choices = ['all', 'forwarding', 'acl']),
+                        clear_group = dict(choices = ['all', 'user', 'auto-export'])
+                        )
+        argument_spec = dict()
+        argument_spec.update(element_spec)
+        self._module = AnsibleModule(
+            argument_spec=argument_spec,
+            supports_check_mode=True)
+
+    def get_required_config(self):
+        self._required_config = dict()
+        module_params = self._module.params
+        group = module_params.get('group')
+        export_group = module_params.get('export_group')
+        clear_group = module_params.get('clear_group')
+
+        auto_export = module_params.get('auto_export', True)
+        enabled = module_params.get('enabled', True)
+
+        params = dict()
+        if group:
+            params.update({
+                'group': group,
+                'enabled': enabled
+            })
+
+        if export_group:
+            params.update({
+                'export_group': export_group,
+                'auto_export': auto_export
+            })
+
+        if clear_group:
+            params.update({
+                'clear_group': clear_group
+            })
+
+        self.validate_param_values(params)
+        self._required_config = params
+
+    def _get_wjh_config(self):
+        return show_cmd(self._module, "show running-config | include .*what-just-happened.*")
+        
+    def _set_current_config(self, config):
+        if not config:
+            return 
+        current_config = self._current_config #alias 
+        for line in config:
+            match = self.WJH_DISABLED_REGX.match(line)
+            if match: 
+                #wjh is disabled for one group
+                group = match.group(1)
+                current_config[group] = False 
+
+            match = self.WJH_DISABLED_AUTO_EXPORT_REGX.match(line)
+            if match:
+                #wjh auto export is disabled for one group
+                export_group = match.group(1) + '_export'
+                current_config[export_group] = False 
+    
+    '''
+        show running config will contains [no wjh * group enable] if disabled - default config is enabled 
+    '''
+    def load_current_config(self):
+        self._current_config = dict()
+        config_lines = self._get_wjh_config().get('Lines', None)
+        
+        if config_lines:
+            self._set_current_config(config_lines)
+
+    '''
+        wjh is enabled "by default"
+        when wjh disable we  won't find wjh commands in running config 
+    ''' 
+    def generate_commands(self):
+        current_config, required_config = self._current_config, self._required_config
+        for key, value in required_config.items():
```

Comment:
Are the keys mutual exclusive?

Reply:
yes I just connect between the value of current config and use keys to indicate what is the required config.